statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,626 @@
+            jdata -= jdata.mean(0)
+        else:
+            raise ValueError("For bootstrap, center by 'est' or 'global'")
+        self.vcov = np.dot(jdata.T, jdata) / n_reps
+        if self.vcov.ndim == 2:
+            self.stderr = np.sqrt(np.diag(self.vcov))
+        else:
+            self.stderr = np.sqrt(self.vcov)
+        return est, self.vcov, self.stderr
+
+    def _jackknife(self):
+        """
+        Jackknife variance estimation for survey data.
+
+        Parameters
+        ----------
+        stat : object
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vcov : square ndarray
+            The variance-covariance matrix of the estimates, obtained using
+            the (drop 1) jackknife procedure.
+        pseudo : ndarray
+            The jackknife pseudo-values.
+        """
+        if self.center_by != 'est' and self.design.rep_weights is not None:
+            raise ValueError("If providing replicate weights, center by est")
+
+        est = self._stat(self.design.weights)
+
+        jdata = []
+        # for each cluster
+        for c in range(self.design.nclust):
+            # get jackknife weights
+            w = self.design.get_rep_weights(c=c, cov_method='jack')
+            jdata.append(self._stat(w))
+        jdata = np.asarray(jdata)
+        if self.center_by == 'est':
+            print('centering by estimate')
+            jdata -= est
+        elif self.center_by == 'global':
+            print("centering by global mean")
+            jdata -= jdata.mean(0)
+        else:
+            print("centering by stratum")
+            if self.design.rep_weights is None:
+                for s in range(self.design.n_strat):
+                    # center the 'delete 1' statistic
+                    jdata[self.design.ii[s], :] -= jdata[self.design.ii[s],
+                                                         :].mean(0)
+            else:
+                raise ValueError("Can not center by stratum when rep_weights given")
+
+        if self.design.rep_weights is None:
+            nh = self.design.clust_per_strat[self.design.strat_for_clust].astype(np.float64)
+            _pseudo = jdata + nh[:, None] * (np.dot(self.design.weights,
+                                                self.data) - jdata)
+            mh = np.sqrt((nh - 1) / nh)
+            fh = np.sqrt(1 - self.design.fpc)
+            jdata = fh[:, None] * mh[:, None] * jdata
+        else:
+            nh = self.design.rep_weights.shape[1]
+            mh = np.sqrt((nh - 1) / nh)
+            jdata *= mh
+        self.vcov = np.dot(jdata.T, jdata)
+        if self.vcov.ndim == 2:
+            self.stderr = np.sqrt(np.diag(self.vcov))
+        else:
+            self.stderr = np.sqrt(self.vcov)
+        return est, self.vcov, self.stderr
```

Comment:
It's strange to return a value and attach it to the class.  It mixes functional and OO styles.